test/json_formatter_SUITE.erl

```diff
@@ -0,0 +1,392 @@
+                      <<"pid">> := _Pid},
+      <<"what">> := <<"chars_are_limited">>,
+      <<"text">> := <<"JSON-match-this-chars-limited">>,
+      <<"long_binary">> := ShortenedBinary,
+      <<"long_text">> := ShortenedText,
+      <<"long_struct">> := ShortenedStruct}
+    = Decoded.
+
+format_depth_limited(Config) ->
+
+    {_, File} = ?config(logger_backend, Config),
+    Before = get_log(File),
+
+    FormatDepth = ?config(format_depth, Config),
+    DeepStruct = deep_tuple(FormatDepth*2),
+
+    ?LOG_INFO(#{what => format_depth_limited,
+                text => "JSON-match-this-struct-depth-limited",
+                deep_struct => DeepStruct}),
+
+    After = case get_at_least_n_log_lines(?LOGFILE,
+                                          length(Before) + 1,
+                                          timer:seconds(1)) of
+                timeout -> ct:fail("timed out waiting for messages to reach the log file");
+                Res -> Res
+            end,
+
+    [Line] = filter_out_non_matching(After -- Before, <<"JSON-match-this-struct-depth-limited">>),
+
+    Decoded = jiffy:decode(Line, [return_maps]),
+
+    ShortenedStruct = unicode:characters_to_binary(io_lib:format("~0P", [DeepStruct, FormatDepth])),
+
+    #{<<"level">> := <<"info">>,
+      <<"when">> := _Time,
+      <<"meta">> := #{<<"file">> := _File,
+                      <<"line">> := _Line,
+                      <<"mfa">> := _MFA,
+                      <<"pid">> := _Pid},
+      <<"what">> := <<"format_depth_limited">>,
+      <<"text">> := <<"JSON-match-this-struct-depth-limited">>,
+      <<"deep_struct">> := ShortenedStruct}
+    = Decoded.
+
+
+json_depth_limited(Config) ->
+
+    {_, File} = ?config(logger_backend, Config),
+    Before = get_log(File),
+
+    Depth = ?config(depth, Config),
+    DeepList = deep_list(Depth*2),
+
+    ?LOG_INFO(#{what => json_depth_limited,
+                text => "JSON-match-this-json-depth-limited",
+                deep_list => DeepList}),
+
+    After = case get_at_least_n_log_lines(?LOGFILE,
+                                          length(Before) + 1,
+                                          timer:seconds(1)) of
+                timeout -> ct:fail("timed out waiting for messages to reach the log file");
+                Res -> Res
+            end,
+
+    [Line] = filter_out_non_matching(After -- Before, <<"JSON-match-this-json-depth-limited">>),
+
+    Decoded = jiffy:decode(Line, [return_maps]),
+
+    % 1 because it is at the first level
+    ShortenedList = deep_list(Depth - 1, <<"...">>),
+
+    #{<<"level">> := <<"info">>,
+      <<"when">> := _Time,
+      <<"meta">> := #{<<"file">> := _File,
+                      <<"line">> := _Line,
+                      <<"mfa">> := _MFA,
+                      <<"pid">> := _Pid},
+      <<"what">> := <<"json_depth_limited">>,
+      <<"text">> := <<"JSON-match-this-json-depth-limited">>,
+      <<"deep_list">> := ShortenedList}
+    = Decoded.
```

Comment:
What about making these an `?assertMatch`? It usually has better error messages when it fails 🤔

Reply:
I wanted to, but sometimes I also used unbounded variables in similar places and it wouldn't compile. but here it should indeed work.